### PR TITLE
10-07-24: Add Axios

### PR DESCRIPTION
**Pull request by MazOneTwoOne**
- promise based HTTP client for the browser and node.js
- make axios make an API call to retrieve some fake users

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,4 +27,39 @@ Then, load http://localhost:3000/ in your browser to access the app.
...
+const router = express.Router();
+
+/* GET home page. */
+router.get('/', (req, res, next) => {
+  res.render('main/index', { title: 'Express' });
+});
+
+// Make an API call with `Axios` and `middleware-axios`
+// GET users from external API
+router.get('/users', async (req, res, next) => {
```
Comment (psweeting1) [inline README.md:53]: This needs to be in the README.md

File: utils/axiosSetp.mjs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { create } from 'middleware-axios';
+
+// create wrapped instance of axios to use normal axios instance
+const axios_api = create({
```
Comment (psweeting1) [inline utils/axiosSetp.mjs:4]: `import { create } from 'middleware-axios';

/**
 * Axios middleware to attach Axios instance to request object.
 *
 * @param {object} req - The Express request object.
 * @param {object} res - The Express response object.
 * @param {function} next - The next middleware function in the stack.
 */
const axiosMiddleware = (req, res, next) => {
    const protocol = req.protocol;
    const host = req.get('host');
    const baseURL = `${protocol}://${host}`;
    
    // create wrapped instance of axios to use normal axios instance
    req.axios = create({
        baseURL: baseURL,
        timeout: 5000, // Set a timeout value if needed
        headers: {
            'Content-Type': 'application/json',
            // You can add other default headers here if needed
        },
    });
    next();
};

export default axiosMiddleware;`

File: .env.example [hunk 3]
```diff
@@ -9,4 +9,5 @@ SERVICE_NAME='SERVICE_NAME'
 SERVICE_PHASE='SERVICE_PHASE'
 SERVICE_URL='SERVICE_URL'
 RATE_LIMIT_MAX=100
-RATE_WINDOW_MS=900000#15 * 60 * 1000 = 900000
\ No newline at end of file
+RATE_WINDOW_MS=900000#15 * 60 * 1000 = 900000
+API_BASE_URL='http://localhost:3000/'
```
Comment (psweeting1) [inline .env.example:13]: This is hard coding the API_BASE_URL to always use localhost. What you need to do is hook into the project hosted url. I've added a code snippet to your middleware code that does this for you.

File: README.md (hunk 1 above)
Comment (MazOneTwoOne) [reply README.md:53]: @psweeting1 - this is the `README.md` ????

File: utils/axiosSetp.mjs (hunk 2 above)
Comment (MazOneTwoOne) [reply utils/axiosSetp.mjs:4]: @psweeting1 - Not sure what you want me to do with this comment. Copying and pasting this in, throws a bunch of errors. I'm using the [middelware_axios ](https://www.npmjs.com/package/middleware-axios), which I think does the above already hence implementing that way, .

File: README.md (hunk 1 above)
Comment (psweeting1) [reply README.md:53]: WHOOPS! That's me. Github was telling me it was in app.mjs for some reason.

File: utils/axiosSetp.mjs (hunk 2 above)
Comment (psweeting1) [reply utils/axiosSetp.mjs:4]: Sorry I'll need the errors in order to help you.

**Review by psweeting1 — APPROVED** [review]
LGTM

**Pull request merged**